pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -226,6 +232,45 @@ public void run() {
     return brokerServerBuilder;
   }
 
+  /**
+   * To be overridden to plug in custom external view change handlers.
+   * <p>NOTE: all change handlers will be run in a single thread, so any slow change handler can block other change
+   * handlers from running. For slow change handler, make it asynchronous.
+   *
+   * @param spectatorHelixManager Spectator Helix manager
+   * @return List of custom external view change handlers to plug in
+   */
+  @SuppressWarnings("unused")
+  protected List<ClusterChangeHandler> getCustomExternalViewChangeHandlers(HelixManager spectatorHelixManager) {
```

Comment:
Why not inject configuration for class names and instantiate them through a factory like we do for other such classes?

Reply:
I'm not a big fan of using Class.newInstance() because of the reasons here: https://stackoverflow.com/questions/4612386/what-is-the-difference-between-the-new-operator-and-class-newinstance
Also, it might be hard to setup handlers with zero-argument constructor if user want to pass in some properties not in open source code.

Currently we have both flavors of code in the code base. We can discuss this offline.

Reply:
I did think along the same lines as @mcvsubbu - however it appeared that the HelixBrokerStarter is "expected" to be extended and hence its cleaner to let it override the method than introduce yet another config and instantiate through reflection.

Reply:
How will someone running pinot-broker.sh use this feature?